contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -57,36 +59,43 @@ contract UmbraBatchSend {
   }
 
   function _batchSendEth(uint256 _tollCommitment, SendEth[] calldata _params) internal {
-    for (uint256 i = 0; i < _params.length; i++) {
+    uint256 length = _params.length;
+    for (uint256 i = 0; i < length;) {
       umbra.sendEth{value: _params[i].amount + _tollCommitment}(_params[i].receiver, _tollCommitment, _params[i].pkx, _params[i].ciphertext);
+      unchecked { i++; }
     }
   }
 
   function _batchSendTokens(uint256 _tollCommitment, SendToken[] calldata _params) internal {
-    for (uint256 i = 0; i < _params.length; i++) {
+    uint256 length = _params.length;
+    for (uint256 i = 0; i < length;) {
       IERC20 token = IERC20(address(_params[i].tokenAddr));
 
       SafeERC20.safeTransferFrom(token, msg.sender, address(this), _params[i].amount);
```

Comment:
@mds1 Ik you've made a couple suggestion for this, but my implementations of them weren't great in previous PRs. Any particular direction you can point me to?

Reply:
Specifically transferring tokens each time

Reply:
Implemented SwapSummary